src/lib.rs

```diff
@@ -327,4 +338,32 @@ mod tests {
         assert_eq!(c.take_or_clone_val(), Some(3));
         assert_eq!(c1.take_or_clone_val(), Some(2));
     }
+
+    #[test]
+    fn test_hash() {
+        fn calculate_hash<T: Hash>(t: &T) -> u64 {
+            let mut s = DefaultHasher::new();
+            t.hash(&mut s);
+            s.finish()
+        }
+
+        let c1 = Cactus::new().child(4).child(3);
+        let c2 = Cactus::new().child(4).child(3);
+        assert_eq!(calculate_hash(&c1), calculate_hash(&c2));
+        // The next test is fragile in theory although probably not in practise. Since there's no
+        // guarantee that two distinct things will map to distinct hashes, it's perfectly possible
+        // that a hasher returns the same value for two distinct cactuses. But this isn't hugely
+        // likely to happen and, if it does, it'll be easy to work out what happened; it therefore
+        // seems better to cope with the more likely "someone temporarily replaced the hasher with
```

Comment:
I was with you until the part about replacing with constants.

Reply:
Maybe I was projecting my development style too much ;) I'll remove that part.

Reply:
I think that what you are saying here is that hashing collisions happen, but rarely.

Reply:
Yes, so I was also trying to motivate why I've written a test which is theoretically fragile. But perhaps I over-motivated. I removed the confusing bit.